### ISSUE TITLE
new_vref does not have enough bound

### ISSUE THREAD

**Issue by stevefan1999-personal** [slint-ui/slint#1366]
Running this code:
```rs
use type_uuid::{Bytes, TypeUuid};
use vtable::*;

#[vtable]
#[repr(C)]
struct IUnknownVTable {
    /// Associated constant.
    UUID_: Bytes,
}

impl<T: type_uuid::TypeUuid> IUnknownConsts for T {
    const UUID_: Bytes = Self::UUID;
}

#[derive(TypeUuid)]
#[uuid = "d4adfc76-f5f4-40b0-8e28-8a51a12f5e46"]
#[repr(C)]
struct MyType;

// The #[vtable] macro also exposed a macro to create a vtable
IUnknownVTable_static!(static MY_TYPE_VT for MyType);


fn main() {
    let mut my_type = MyType{};

    new_vref!(let my_vref : VRef<IUnknownVTable> for IUnknown = &my_type);

    println!("{:?}", my_vref.get_vtable().UUID_);
}

```

Output:
```
   Compiling vtable_uuid v0.1.0 (C:\Users\steve\rust\vtable_uuid)
warning: unused import: `#[vtable]`                                                                                                                                                                                                                                                                                    
  --> src\main.rs:4:1
   |
4  | #[vtable]
   | ^^^^^^^^^
...
21 | IUnknownVTable_static!(static MY_TYPE_VT for MyType);
   | ---------------------------------------------------- in this macro invocation
   |
   = note: `#[warn(unused_imports)]` on by default
   = note: this warning originates in the macro `IUnknownVTable_static` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `MyType: IUnknown_vtable_mod::IUnknown` is not satisfied                                                                                                                                                                                                                                 
  --> src\main.rs:27:65
   |
27 |     new_vref!(let my_vref : VRef<IUnknownVTable> for IUnknown = &my_type);
   |     ------------------------------------------------------------^^^^^^^^-
   |     |                                                           |
   |     |                                                           the trait `IUnknown_vtable_mod::IUnknown` is not implemented for `MyType`
   |     required by a bound introduced by this call
   |
note: required by a bound in `get_vt`
  --> src\main.rs:27:5
   |
27 |     new_vref!(let my_vref : VRef<IUnknownVTable> for IUnknown = &my_type);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get_vt`
   = note: this error originates in the macro `new_vref` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `X: TypeUuid` is not satisfied
  --> src\main.rs:27:5
   |
27 |     new_vref!(let my_vref : VRef<IUnknownVTable> for IUnknown = &my_type);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TypeUuid` is not implemented for `X`
   |
note: required because of the requirements on the impl of `IUnknown_vtable_mod::IUnknownConsts` for `X`
  --> src\main.rs:11:30
   |
11 | impl<T: type_uuid::TypeUuid> IUnknownConsts for T {
   |                              ^^^^^^^^^^^^^^     ^
note: required by a bound in `IUnknown_vtable_mod::IUnknownVTable::new`
  --> src\main.rs:6:8
   |
4  | #[vtable]
   | --------- required by a bound in this
5  | #[repr(C)]
6  | struct IUnknownVTable {
   |        ^^^^^^^^^^^^^^ required by this bound in `IUnknown_vtable_mod::IUnknownVTable::new`
   = note: this error originates in the macro `new_vref` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `X`
  --> C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\vtable-0.1.7\src\lib.rs:423:25
   |
423|             fn get_vt<X: type_uuid::TypeUuid $trait_>(_: &X) -> <$vtable as VTableMeta>::VTable {
   |                          +++++++++++++++++++

For more information about this error, try `rustc --explain E0277`.
warning: `vtable_uuid` (bin "vtable_uuid") generated 1 warning
error: could not compile `vtable_uuid` due to 2 previous errors; 1 warning emitted
```

## Context

I was trying to automatically derive a UUID vtable for each object so I can resolve them in runtime for hot swapping. However, it seems like it doesn't account for "extension traits", and so the best way I can do is to resolute to using method invocation (Haven't tried but my guts feeling tells me that would work)

**Comment by ogoffart** [comment]
> the trait `IUnknown_vtable_mod::IUnknown` is not implemented for `MyType`

That error is correct.  you did not implement the trait IUnknown for my MyType, even if it is empty in this case.
Can you try to add

```rust
impl IUnknown for MyType{}
```

**Comment by ogoffart** [comment]
Did you manage to get it work to your liking? Can this issue be closed?

**Comment by stevefan1999-personal** [comment]
@ogoffart yes.i switched it to vfunc based and it should only cost one indirection call. this however should be enough for my rpc use
